numpy/lib/npyio.py

```diff
@@ -599,13 +611,41 @@ def savez_compressed(file, *args, **kwds):
     """
     _savez(file, args, kwds, True)
 
+@contextlib.contextmanager
```

Comment:
Isn't this also a reasonable place to use [`contextlib.closing`](https://docs.python.org/2.6/library/contextlib.html#contextlib.closing), instead of creating your own class? This seems like exactly the situation that function / decorator exists...

(PS sorry for the double comment, the last one was in the wrong place)